lib/pure/algorithm.nim

```diff
@@ -352,7 +352,7 @@ proc product*[T](x: openArray[seq[T]]): seq[seq[T]] =
     index = 0
     indexes[index] -= 1
 
-proc nextPermutation*[T](x: var openarray[T]): bool {.discardable.} =
+proc nextPermutation*[T](x: var openarray[T]): bool {.discardable,inline.} =
```

Comment:
Do you really have programs that call ``nextPermutation`` from 10 different call sites where you worry about that 10X amplification?  Or is this just a heuristic/theoretical gut feeling?  The Nim code there goes almost straight-line over to mirrored C code and is not very big in practical terms and permutations explode fast enough that sometimes this can make a big performance impact.

Reply:
We don't know the ``T``, so I agree with mratsim, these two should not become ``inline`` (which is generally overused, rather than underused, IMO). Link time optimizations can inline every proc but optimizers are bad at extracting procs back from inlined code.